src/scheduler/fifo.c

```diff
@@ -1975,6 +1975,8 @@ add_job_to_calendar(int pbs_sd, status *policy, server_info *sinfo,
 					create_node_array_from_nspec(bjob->nspec_arr);
 				selectspec = create_select_from_nspec(bjob->nspec_arr);
 				if (selectspec != NULL) {
+					if (bjob->execselect)
```

Comment:
Thanks for fixing this leak.
I don't think you need to check for NULL here.

Reply:
ok. removed the check.